### DISCUSSION TITLE
世纪互联的SharePoint部署不管userPrincipalName填什么都提示并非网站所有者

### DISCUSSION THREAD

**Discussion by iskycc** [spencerwooo/onedrive-vercel-index#479] [Q&A] [answered]
如题，我在userPrincipalName里填了我能想到的所有可能，但是没有一个能成功

**Comment by iskycc** [comment]
请问正常情况下，部署SharePoint的话，userPrincipalName应该填什么呢？

**Comment by spencerwooo** [comment]
**你在这么多渠道（多个 discussion，多个 issue，多封 email，甚至在无关的 Telegram 群里）问我，有没有考虑过我在忙没时间回复你呢？这样很不礼貌。**

关于你的问题，在：

<pre>
<a href="https://developer.microsoft.com/en-us/graph/graph-explorer">https://developer.microsoft.com/en-us/graph/graph-explorer</a>
</pre>

用你的账户登录，然后 run query，看默认 `https://graph.microsoft.com/v1.0/me` API 路径给的 `userPrincipalName` 是什么。

**Reply by iskycc** [reply]
不好意思，关于打扰你非常抱歉，我试了很多没有找到解决办法，而且世纪互联的API好像和国际版的不通用

**Reply by iskycc** [reply]
如果通用的就不会出问题，我花了一下午的时间去尝试

**Reply by iskycc** [reply]
另外，我只在一个 discussion，一个 issue，两封 email（其中一封email是网络问题我以为发不出去了，所以我再发了一封）

**Reply by spencerwooo** [reply]
https://ovi.swo.moe/docs/getting-started#modify-api-settings-optional

**Reply by iskycc** [reply]
> https://ovi.swo.moe/docs/getting-started#modify-api-settings-optional

我早就已经认真看完了这个文档，所有的都是正常通过的，问题就出在最后userPrincipalName我填什么都提示我不是网站所有者
![image](https://user-images.githubusercontent.com/81439128/154935292-b4204f69-3170-407d-acb7-6bcf736981cd.png)

**Reply by zkeq** [reply]
你的仓库地址方便发一下吗？
或者说 发一下你的 `site.config.js`

**Reply by iskycc** [reply]
> 你的仓库地址方便发一下吗？ 或者说 发一下你的 `site.config.js`

请查看 https://github.com/212-mei/openwrt

**Reply by spencerwooo** [reply]
https://github.com/212-mei/openwrt/blob/05a30a104ce35a61cff35b4013ab75a0eecb7271/config/api.config.js#L23

这是啥？你确定是这个 API 地址？

**Reply by iskycc** [reply]
> https://github.com/212-mei/openwrt/blob/05a30a104ce35a61cff35b4013ab75a0eecb7271/config/api.config.js#L23
> 
> 这是啥？你确定是这个 API 地址？

SharePoint的网址就是那样的，我部署过之前那个版本的项目，用的就是这个drive api，是用https://github.com/beetcb/ms-graph-cli 这个工具获取的，之前就是正常的

**Comment by spencerwooo** [comment]
首先，在这里认证的：

https://github.com/212-mei/openwrt/blob/05a30a104ce35a61cff35b4013ab75a0eecb7271/utils/oAuthHandler.ts#L88-L95

一般认证 URL（就是获取用户 userPrincipleName 的 URL）是：

```
https://graph.microsoft.com/v1.0/me
```

获取文件列表的 API 是：

```
https://graph.microsoft.com/v1.0/me/drive
```

这是国际版、E5 开发者版、以及 OneDrive for Business 的通用 API。世纪互联 SharePoint 确实不一样，但是没想到的是 SharePoint 你这个跟别的 SharePoint 还不一样，所以你到底是什么版本的世纪互联？

**Reply by spencerwooo** [reply]
还是说你的是 SharePoint 站点，不是 Drive？二者应该是不一样的。

**Reply by iskycc** [reply]
对的，我说的就是SharePoint站点

**Comment by spencerwooo** [comment, marked as answer]
我看到你的 API 返回里有 `name`，你可以把这一行的判断条件改了暂时解决问题：

https://github.com/212-mei/openwrt/blob/05a30a104ce35a61cff35b4013ab75a0eecb7271/pages/onedrive-vercel-index-oauth/step-3.tsx#L58

```diff
- if (data.userPrincipalName !== siteConfig.userPrincipalName) {
+ if (data.name !== siteConfig.name) {
```

当然你得在 `siteConfig` 里再加一个 `name` 的字段。

但我不保证后续文件 API 能直接迁移。

**Reply by iskycc** [reply]
了解了，但是我有个思路：既然这样会出现问题，为什么不写个密码放到环境变量里面，然后直接在引导的时候输入密码呢？

**Reply by iskycc** [reply]
问题已经解决了，感谢您的帮助，对你造成的打扰我深感抱歉
